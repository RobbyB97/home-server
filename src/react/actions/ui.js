export const navToggle = () => ({
    type: 'NAV_TOGGLE'
})


export const navOff = () => ({
    type: "NAV_OFF"
})


export const whiteOn = () => ({
    type: "WHITE_ON"
})


export const whiteOff = () => ({
    type: "WHITE_OFF"
})