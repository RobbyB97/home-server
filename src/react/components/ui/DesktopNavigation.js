import React from 'react';
import { connect } from 'react-redux';


/* Component */
export class DesktopNavigation extends React.Component {
    constructor(props) {
        super(props);
    }

    render() {
        return (
            <div id="DesktopNavigation"
                className={this.props.white ?
                    "white":""}
            >
                <ul>
                    <li>
                        <a href="#about">about</a>
                    </li>
                    <li>
                        <a href="#services">services</a>
                    </li>
                    <li>
                        <a href="#skills">skills</a>
                    </li>
                    <li>
                        <a href="#experience">experience</a>
                    </li>
                    <li>
                        <a href="#projects">projects</a>
                    </li>
                </ul>
            </div>
        );
    }
}


const mapStateToProps = (state) => {
    return {
        white: state.ui.white
    }
}


export default connect(mapStateToProps, undefined)(DesktopNavigation)