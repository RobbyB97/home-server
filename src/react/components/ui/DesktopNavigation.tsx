export {}

/* Libraries */
import * as React from 'react'


/* Component */
export class DesktopNavigation extends React.Component<{}, {}> {
    public render() {
        return (
            <div id="DesktopNavigation">
                <ul>
                    <li>
                        <a href="#about">about</a>
                    </li>
                    <li>
                        <a href="#services">services</a>
                    </li>
                    <li>
                        <a href="#skills">skills</a>
                    </li>
                    <li>
                        <a href="#experience">experience</a>
                    </li>
                    <li>
                        <a href="#projects">projects</a>
                    </li>
                </ul>
            </div>
        )
    }
}


export default DesktopNavigation;