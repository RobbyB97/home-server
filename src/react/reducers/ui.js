const UIDefaultState = {
    nav: false,
    white: false
}


export default (state = UIDefaultState, action) => {
    switch(action.type) {
        case 'NAV_TOGGLE':
            return {
                ...state,
                nav: state.nav ? false : true
            }

        case 'NAV_OFF':
            return {
                ...state,
                nav: false
            }

        case 'WHITE_ON':
            return {
                ...state,
                white: true
            }

        case 'WHITE_OFF':
            return {
                ...state,
                white: false
            }

        default:
            return state
    }
}