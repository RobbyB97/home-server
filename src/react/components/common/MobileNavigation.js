import React from 'react'
import {NavLink} from 'react-router-dom'
import { connect } from 'react-redux'

import {toggleNav, toggleNavOff} from '../../actions/navigation'
import {setTitle} from '../../actions/title'
import {homeOn, homeOff} from '../../actions/home'


const MobileNavigation = ({nav, dispatch, home}) => (
    <div id="mobile-navigation" className={nav ? 'active-nav':''}>
        <div className="mobile-navigation__background">
            <ul className="mobile-navigation__intra-nav">
                <li>
                    <NavLink to="/" onClick={() => {
                        dispatch(toggleNav())
                        dispatch(setTitle(''))
                        dispatch(homeOn())
                    }}>Home</NavLink>
                </li>

                <li>
                    <NavLink to="/skills/" onClick={() => {
                        dispatch(toggleNav())
                        dispatch(setTitle('Skills'))
                        dispatch(homeOff())
                    }}>Skills</NavLink>
                </li>

                <li>
                    <NavLink to="/experience/" onClick={() => {
                        dispatch(toggleNav())
                        dispatch(setTitle('Experience'))
                        dispatch(homeOff())
                    }}>Experience</NavLink>
                </li>

                <li>
                    <NavLink to="/projects/" onClick={() => {
                        dispatch(toggleNav())
                        dispatch(setTitle('Projects'))
                        dispatch(homeOff())
                    }}>Projects</NavLink>
                </li>

                <li>
                    <NavLink to="/contact/" onClick={() => {
                        dispatch(toggleNav())
                        dispatch(setTitle('Contact'))
                        dispatch(homeOff())
                    }}>Contact</NavLink>
                </li>
            </ul>
        </div>
    </div>
)

const mapStateToProps = (state) => {
    return {
        nav: state.navigation.nav,
        home: state.home.home
    }
}

export default connect(mapStateToProps)(MobileNavigation)