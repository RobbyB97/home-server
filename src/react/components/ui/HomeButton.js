/* Libraries */
import React from 'react'
import {NavLink} from 'react-router-dom'
import {connect} from 'react-redux'

import {navToggle} from '../../actions/ui'


/* Component */
export class HomeButton extends React.Component {
    constructor(props) {
        super(props);
        this.state = {
            show: false
        }
    }

    componentDidMount() {
        const component = this;
        document.addEventListener('scroll', () => {
            component.toggle()
        })
    }

    toggle = () => {
        if (window.pageYOffset > 300) {
            this.setState({
                show: true
            })
        } else {
            this.setState({
                show: false
            })
        }
    }

    navToggle = () => {
        this.props.navToggle()
    }

    render() {
        return (
            <div id="HomeButton" 
                className={this.props.nav || this.props.white ? "nav":""}
                onClick={this.props.nav && this.navToggle}
                >
                
                {this.props.internal ?
                    <a href="#app" className={this.state.show ?
                        "homeButton__logo show":
                        "homeButton__logo"
                    }>B</a>
                :
                    <NavLink to="/" onClick={this.toTop}
                        className={this.state.show ?
                            "homeButton__logo show":
                            "homeButton__logo"
                        }>
                        B
				    </NavLink>
                }
            </div>
        )
    }
}


/* Connect to store */
const mapStateToProps = (state) => {
    return {
        nav: state.ui.nav,
        white: state.ui.white
    }
}

const mapDispatchToProps = (dispatch) => ({
    navToggle: () => {
        dispatch(navToggle())
    }
})


/* Export */
export default connect(mapStateToProps, mapDispatchToProps)(HomeButton)