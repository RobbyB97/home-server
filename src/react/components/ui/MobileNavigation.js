/* Libraries */
import React from 'react'
import {connect} from 'react-redux'

import {navToggle} from '../../actions/ui'


/* Component */
export class MobileNavigation extends React.Component {
    constructor(props) {
        super(props);
    }

    navToggle = () => {
        this.props.navToggle();
    }

    render() {
        return (
            <div id="MobileNavigation" 
                className={this.props.white ?
                    "white":""}
            >
                <div className={this.props.nav ?
                    "mobileNavigation__button nav":
                    "mobileNavigation__button"}>
                    <div onClick={this.navToggle}>
					    {this.props.nav ? '✕':'☰'}
                    </div>
				</div>

                <div className={this.props.nav ? 
                    "mobileNavigation__menu nav":
                    "mobileNavigation__menu"}>
                    <ul className="mobileNavigation__links">
                        <li>
                            <a href="#about" onClick={this.navToggle}>about</a>
                        </li>
                        <li>
                            <a href="#services" onClick={this.navToggle}>services</a>
                        </li>
                        <li>
                            <a href="#skills" onClick={this.navToggle}>skills</a>
                        </li>
                        <li>
                            <a href="#experience"onClick={this.navToggle}>experience</a>
                        </li>
                        <li>
                            <a href="#projects" onClick={this.navToggle}>projects</a>
                        </li>
                    </ul>
                </div>
            </div>
        )
    }
}


/* Connect to store */
const mapStateToProps = (state) => {
    return {
        nav: state.ui.nav,
        white: state.ui.white
    }
}

const mapDispatchToProps = (dispatch) => ({
    navToggle: () => {
        dispatch(navToggle())
    }
})


/* Export */
export default connect(mapStateToProps, mapDispatchToProps)(MobileNavigation)