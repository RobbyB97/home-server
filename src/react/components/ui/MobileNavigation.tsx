export {}

/* Libraries */
import * as React from 'react';
import {connect} from 'react-redux';

import {navToggle} from '../../actions/ui';


/* Component */
interface Props {
    nav: boolean,
    navToggle: typeof navToggle
}
interface State {
    nav: boolean,
    navToggle: typeof navToggle
}

export class MobileNavigation extends React.Component<Props, State> {
    constructor(props: any) {
        super(props);
        this.state = {
            ...this.state
        }
    }

    navToggle = () => {
        this.state.navToggle();
    };

    public render() {
        const {nav} = this.state;
        const {navToggle} = this.state;

        return (
            <div id="MobileNavigation">
                <div className={nav ?
                    "mobileNavigation__button nav":
                    "mobileNavigation__button"}>
                    <div onClick={navToggle}>
					    {nav ? '✕':'☰'}
                    </div>
				</div>

                <div className={nav ? 
                    "mobileNavigation__menu nav":
                    "mobileNavigation__menu"}>
                    <ul className="mobileNavigation__links">
                        <li>
                            <a href="#about" onClick={navToggle}>about</a>
                        </li>
                        <li>
                            <a href="#services" onClick={navToggle}>services</a>
                        </li>
                        <li>
                            <a href="#skills" onClick={navToggle}>skills</a>
                        </li>
                        <li>
                            <a href="#experience"onClick={navToggle}>experience</a>
                        </li>
                        <li>
                            <a href="#projects" onClick={navToggle}>projects</a>
                        </li>
                    </ul>
                </div>
            </div>
        );
    }
}

/* Connect to store */
const mapStateToProps = (state) => ({
    nav: state.ui.nav,
});

const mapDispatchToProps = (dispatch) => ({
    navToggle: () => {
        dispatch(navToggle())
    }
});


/* Export */
export default connect(mapStateToProps, mapDispatchToProps)(MobileNavigation);