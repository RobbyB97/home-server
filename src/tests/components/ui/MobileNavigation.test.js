import React from 'react'
import {shallow} from 'enzyme'

import {MobileNavigation} from '../../../react/components/ui/MobileNavigation'


test('Render MobileNavigation', () => {
    const wrapper = shallow(<MobileNavigation />)
    expect(wrapper).toMatchSnapshot()
})