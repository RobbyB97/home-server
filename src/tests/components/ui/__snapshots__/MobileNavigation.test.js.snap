// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render MobileNavigation 1`] = `
<div
  className=""
  id="MobileNavigation"
>
  <div
    className="mobileNavigation__button"
  >
    <div
      onClick={[Function]}
    >
      â˜°
    </div>
  </div>
  <div
    className="mobileNavigation__menu"
  >
    <ul
      className="mobileNavigation__links"
    >
      <li>
        <a
          href="#about"
          onClick={[Function]}
        >
          about
        </a>
      </li>
      <li>
        <a
          href="#services"
          onClick={[Function]}
        >
          services
        </a>
      </li>
      <li>
        <a
          href="#skills"
          onClick={[Function]}
        >
          skills
        </a>
      </li>
      <li>
        <a
          href="#experience"
          onClick={[Function]}
        >
          experience
        </a>
      </li>
      <li>
        <a
          href="#projects"
          onClick={[Function]}
        >
          projects
        </a>
      </li>
    </ul>
  </div>
</div>
`;
